### PR TITLE
adds the manufacturing user

### PR DESCRIPTION
**Pull request by mc-oofert**
## About The Pull Request

https://github.com/user-attachments/assets/2c60d47b-1671-41ca-a940-f06ea497d626

the manufacturing user is a built with a slab of meat and a servo
it will attempt to use its held object on whatever is infront of it then send it forwards after doing so
it may receive a new held object via conveyor
you can open its panel to **feed it a stray cat** (or any animal) or any actual food item for a temporary boost in doafter speed
cannot attack hopefully
more reliable code than a big manipulator (skill issue!!!)

runtime log during video
2024-11-07 10_58_06-Window

## Why It's Good For The Game

more factory automation stuff for engineers to do
gif

## Changelog
:cl:
add: added the manufacturing user
/:cl:

### CODE REVIEW

**Comment by SmArtKar** [comment]
We *really* shouldn't be using abstract mobs inside of machines to use tools. If big manipulators taught us anything, that'd be it.

**Comment by LemonInTheDark** [comment]
if you cannot do it sanely, do not do it. Especially if you have such a nonspecific sell

**Comment by mc-oofert** [comment]
its as sane as it gets

**Comment by LemonInTheDark** [comment]
no it is not

**Comment by mc-oofert** [comment]
> We _really_ shouldn't be using abstract mobs inside of machines to use tools. If big manipulators taught us anything, that'd be it. Its nullspaced too, which is _nasty_

big manipulators taught us that maybe we shouldnt create a new mob every single interaction

**Comment by mc-oofert** [comment]
idk ive tested this thing and it has not runtimed or shit the bed yet

File: code/modules/manufactorio/machines/user.dm [hunk 1]
```diff
@@ -0,0 +1,233 @@
+// dummy carbon to do item stuff for machines
+/mob/living/carbon/machine_user
```
Comment (optimumtact) [inline code/modules/manufactorio/machines/user.dm:2]: lol said the scorpion
lmao said the frog

**Comment by SmArtKar** [comment]
actually potato changed my mind

**Comment by mc-oofert** [comment]
guess ill have to pr it to downstreams instead then ig

**Comment by Time-Green** [comment]
or refactor item attack to not require a mob 🥹

**Comment by MrMelbert** [comment]
it just needs to be a basic mob with an ai, it's that simple

**Comment by mc-oofert** [comment]
> it just needs to be a basic mob with an ai, it's that simple

basicmobs suck shit im going to be real

**Comment by mc-oofert** [comment]
> We _really_ shouldn't be using abstract mobs inside of machines to use tools. If big manipulators taught us anything, that'd be it. Its nullspaced too, which is _nasty_

monke does it
https://github.com/Monkestation/Monkestation2.0/blob/24e9763c3e8c19c181498eafa6e4113a67aee5c1/monkestation/code/modules/mech_comp/objects/interactor.dm

**Comment by LemonInTheDark** [comment]
respectfully, this means absolutely nothing and only serves as an indictment of monke's maintainers

**Comment by mc-oofert** [comment]
> respectfully, this means absolutely nothing and only serves as an indictment of monke's maintainers

what, that they're seemingly prioritizing fun > code quality?
(their codebase is still a weird mess of stupid PRs and 200000 species don't get me wrong though)

**Comment by MrMelbert** [comment]
It's not even a "code quality thing", it's just that this kind of stuff is so fragile that it's basically guaranteed to break mid to long term. (I mean the last iteration of this broke instantly) 

This PR is even more cursed than the first iteration because it branches off carbon instead of human which we haven't done since like 2017

**Comment by optimumtact** [comment]
> what, that they're seemingly prioritizing fun > code quality? (their codebase is still a weird mess of stupid PRs and 200000 species don't get me wrong though)

Respectfully this gets them so far, but there's a reason downstreams like this constantly abandon their entire codebase to rebuild of tg every time.

**Pull request closed without merging**